src/main/kotlin/uk/co/craigbass/pratura/boundary/AddShippingAddress.kt

```diff
@@ -0,0 +1,17 @@
+package uk.co.craigbass.pratura.boundary
+
+import com.madetech.clean.usecase.SynchronousUseCase
+import uk.co.craigbass.pratura.boundary.AddShippingAddress.Request
+
+interface AddShippingAddress : SynchronousUseCase<Request, Unit> {
+  data class Request(
+    val name: String,
+    val companyName: String?,
+    val addressLine1: String,
+    val addressLine2: String?,
```

Comment:
A NonEmptyList type probably exists either in Kotlin or is trivial to make

But as nice as the multi address line model is it'll probably give you headaches interoperating with third parties that use the standard line1/line2 deal